services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_pipelines.py

```diff
@@ -16,14 +16,14 @@
 
 class CompPipelinesRepository(BaseRepository):
     @log_decorator(logger=logger)
-    async def publish_pipeline(
-        self, project_id: ProjectID, dag_graph: nx.DiGraph
+    async def upsert_pipeline(
```

Comment:
much better ... ``publish`` is too ambiguous while ``upsert`` is super concrete!